internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,246 @@
+	netAppValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NetAppVolumeQuotaRuleResource struct{}
+
+var _ sdk.Resource = NetAppVolumeQuotaRuleResource{}
+
+func (r NetAppVolumeQuotaRuleResource) ModelObject() interface{} {
+	return &netAppModels.NetAppVolumeQuotaRuleModel{}
+}
+
+func (r NetAppVolumeQuotaRuleResource) ResourceType() string {
+	return "azurerm_netapp_volume_quota_rule"
+}
+
+func (r NetAppVolumeQuotaRuleResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumequotarules.ValidateVolumeQuotaRuleID
+}
+
+func (r NetAppVolumeQuotaRuleResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeQuotaRuleName,
+		},
+
+		"location": commonschema.Location(),
+
+		"volume_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.VolumeID,
+		},
+
+		"quota_type": {
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			Required:     true,
+			ValidateFunc: validation.StringInSlice(volumequotarules.PossibleValuesForType(), false),
+		},
+
+		"quota_target": {
+			Type:         pluginsdk.TypeString,
+			ForceNew:     true,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"quota_size_in_kib": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(4),
+		},
+	}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeQuotaRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			volumeID, err := volumes.ParseVolumeID(model.VolumeID)
+			if err != nil {
+				return fmt.Errorf("error parsing volume id %s: %+v", model.VolumeID, err)
+			}
+
+			id := volumequotarules.NewVolumeQuotaRuleID(subscriptionId, volumeID.ResourceGroupName, volumeID.NetAppAccountName, volumeID.CapacityPoolName, volumeID.VolumeName, model.Name)
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.Get(ctx, id)
+			if err != nil && existing.HttpResponse.StatusCode != http.StatusNotFound {
```

Comment:
These [lines](https://github.com/hashicorp/terraform-provider-azurerm/pull/23042/files#diff-173ac1cb46340a0d51c85237298e707803ac534f4f4dbcb8e056a77d215124f6R106-R112) can be consolidated into something like 

```
			if err != nil {
				if !response.WasNotFound(existing.HttpResponse) {
					return fmt.Errorf("checking for presence of existing %s: %s", id, err)
				}
			}

			if !response.WasNotFound(existing.HttpResponse) {
				return tf.ImportAsExistsError(r.ResourceType(), id.ID())
			}
```